test/webgpu_basic_rendering.cpp

```diff
@@ -383,15 +391,21 @@ void run() {
 
         wgpu::SurfaceDescriptor surfDesc{};
         surfDesc.nextInChain = &canvasDesc;
-        wgpu::Surface surface = instance.CreateSurface(&surfDesc);
-
-        wgpu::SwapChainDescriptor scDesc{};
-        scDesc.usage = wgpu::TextureUsage::RenderAttachment;
-        scDesc.format = wgpu::TextureFormat::BGRA8Unorm;
-        scDesc.width = kWidth;
-        scDesc.height = kHeight;
-        scDesc.presentMode = wgpu::PresentMode::Fifo;
-        swapChain = device.CreateSwapChain(surface, &scDesc);
+        surface = instance.CreateSurface(&surfDesc);
+
+        wgpu::SurfaceCapabilities capabilities;
+        surface.GetCapabilities(adapter, &capabilities);
+
+        wgpu::SurfaceConfiguration config{
+            .device = device,
+            .format = capabilities.formats[0],
+            .usage = wgpu::TextureUsage::RenderAttachment,
+            .alphaMode = capabilities.alphaModes[0],
+            .width = kWidth,
+            .height = kHeight,
+            // .presentMode = capabilities.presentModes[0]};
+            .presentMode = wgpu::PresentMode::Fifo};
```

Comment:
Should always use `Fifo` (it's guaranteed to be available).
```suggestion
            .presentMode = wgpu::PresentMode::Fifo};
```